pkg/apis/security/v1beta1/types.go

```diff
@@ -0,0 +1,140 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1beta1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type ClusterNetworkPolicy struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard metadata of the object.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Specification of the desired behavior of ClusterNetworkPolicy.
+	Spec ClusterNetworkPolicySpec `json:"spec"`
+}
+
+// ClusterNetworkPolicySpec defines the desired state for ClusterNetworkPolicy.
+type ClusterNetworkPolicySpec struct {
+	// Priority specfies the order of the ClusterNetworkPolicy relative to
+	// other ClusterNetworkPolicies.
+	Priority int32 `json:"priority"`
+	// Select workloads on which the rules will be applied to.
+	AppliedTo []NetworkPolicyPeer `json:"appliedTo"`
+	// Set of ingress rules evaluated based on the order in which they are set.
+	// Currently Ingress rule supports setting the `From` field but not the `To`
+	// field within a Rule.
+	// +optional
+	Ingress []Rule `json:"ingress"`
+	// Set of egress rules evaluated based on the order in which they are set.
+	// Currently Egress rule supports setting the `To` field but not the `From`
+	// field within a Rule.
+	// +optional
+	Egress []Rule `json:"egress"`
+}
+
+// Rule describes the traffic allowed to/from the workloads selected by
+// Spec.AppliedTo. Based on the action specified in the rule, traffic is either
+// allowed or denied which exactly match the specified ports and protocol.
+type Rule struct {
+	// Action specifies the action to be applied on the rule. Defaults to
+	// ALLOW action.
+	// +optional
+	Action *RuleAction `json:"action"`
+	// Set of port and protocol allowed/denied by the rule. If this field is unset
+	// or empty, this rule matches all ports.
+	// +optional
+	Ports []NetworkPolicyPort `json:"ports"`
+	// Rule is matched if traffic originates from workloads selected by
+	// this field. If this field is empty, this rule matches all sources.
+	// +optional
+	From []NetworkPolicyPeer `json:"from"`
+	// Rule is matched if traffic is intended for workloads selected by
+	// this field. If this field is empty or missing, this rule matches all
+	// destinations.
+	// +optional
+	To []NetworkPolicyPeer `json:"to"`
+}
+
+// NetworkPolicyPeer describes the grouping selector of workloads.
+type NetworkPolicyPeer struct {
+	// IPBlock describes the IPAddresses/IPBlocks that is matched in to/from.
```

Comment:
"IPBlock describes the IPAddresses/IPBlocks"?

Not sure how IPBlock can describe the IPBlocks :)
maybe use: IPBlock specifies an IP CIDR

Reply:
IPBlock is self explanatory no? .. updated :)